### PR TITLE
[SYCL][L0] Make all L0 events have device-visibility

### PR DESCRIPTION
**Pull request by smaslov-intel**
Make all L0 events have device-visibility by default, and only create/signal/query their host-visible proxy event on demand.
Since this is not showing better performance yet and requires further tuning, I am going to disable it by default initially.
Test is coming in https://github.com/intel/llvm-test-suite/pull/456

Signed-off-by: Sergey V Maslov <sergey.v.maslov@intel.com>

### CODE REVIEW

**Review by againull — APPROVED** [review]

File: sycl/plugins/level_zero/pi_level_zero.cpp [hunk 1]
```diff
@@ -181,6 +181,14 @@ static void zePrint(const char *Format, ...) {
   }
 }
 
+// Controls whether device-scope events are used.
+static const bool ZeAllHostVisibleEvents = [] {
+  const auto DeviceEventsStr =
+      std::getenv("SYCL_PI_LEVEL_ZERO_DEVICE_SCOPE_EVENTS");
```
Comment (asudarsa) [inline sycl/plugins/level_zero/pi_level_zero.cpp:187]: Can we name this as SYCL_PI_LEVEL_ZERO_ALL_HOST_VISIBLE_EVENTS?

**Review by asudarsa — APPROVED** [review]
This is just a nitpick. Looks good to me..

**Review by bader — APPROVED** [review]
LGTM, but I'd like @pvchupin to approve.

File: sycl/doc/EnvironmentVariables.md [hunk 2]
```diff
@@ -34,6 +34,7 @@ subject to change. Do not rely on these variables in production code.
...
 | `SYCL_PI_LEVEL_ZERO_USE_COPY_ENGINE_FOR_D2D_COPY` (experimental) | Integer | Allows the use of copy engine, if available in the device, in Level Zero plugin for device to device copy operations. The default is 0. This option is experimental and will be removed once heuristics are added to make a decision about use of copy engine for device to device copy operations. |
 | `SYCL_PI_LEVEL_ZERO_TRACK_INDIRECT_ACCESS_MEMORY` | Any(\*) | Enable support of the kernels with indirect access and corresponding deferred release of memory allocations in the Level Zero plugin. |
+| `SYCL_PI_LEVEL_ZERO_DEVICE_SCOPE_EVENTS` | Any(\*) | Enable support of device-scope events whose state is not visible to the host. If enabled the Level Zero plugin would create all events having device-scope only and create proxy host-visible events for them when their status is needed (wait/query) on the host. The default is 0, meaning all events are host-visible. |
```
Comment (pvchupin) [inline sycl/doc/EnvironmentVariables.md:37]: @alexbatashev started cleanup of our environment variables list. We need to categorize this properly whether it's experimental or for production.

**Review by pvchupin — APPROVED** [review]
Ok to me. Please see comment from @asudarsa.

File: sycl/doc/EnvironmentVariables.md (hunk 2 above)
Comment (smaslov-intel) [reply sycl/doc/EnvironmentVariables.md:37]: the intention is to have this user-visible

File: sycl/plugins/level_zero/pi_level_zero.cpp (hunk 1 above)
Comment (smaslov-intel) [reply sycl/plugins/level_zero/pi_level_zero.cpp:187]: We can, but I like current name more because it conveys Level-Zero event "scope" concept

**Pull request merged**